source/public/scripts/main.journal.js

```diff
@@ -0,0 +1,127 @@
+document.addEventListener("DOMContentLoaded", function () {
+    // Select the modal and various buttons
+    var modal = document.getElementById("task-modal");
+    var btn = document.querySelector(".new-task");
+    var saveBtn = document.querySelector(".save-task");
+    var cancelBtn = document.querySelector(".cancel-task");
+    var deleteBtnModal = document.querySelector(".delete-task");
+    var taskContainer = document.getElementById("task-container");
+
+    // Initialize editingIndex to -1 to indicate no task is being edited
+    let editingIndex = -1;
+
+    // Show the modal when the "New Task" button is clicked
+    btn.onclick = function () {
+      editingIndex = -1; // Reset editing index when adding a new task
+      modal.style.display = "block"; // Show the modal
+      clearModalFields(); // Clear any existing data in the modal fields
+    }
+
+    // Hide the modal if the user clicks outside of it
+    window.onclick = function (event) {
+      if (event.target == modal) {
+        modal.style.display = "none";
+      }
+    }
+
+    // Save the task when the "Save" button is clicked
+    saveBtn.onclick = function () {
+      // Get task details from modal input fields
+      var taskDesc = document.getElementById("task-desc").value;
+      var taskType = document.getElementById("task-type").value;
+      var taskProject = document.getElementById("task-project").value;
+
+      // Retrieve existing tasks from local storage or initialize as empty array
+      var tasks = JSON.parse(localStorage.getItem("tasks")) || [];
+
+      if (editingIndex >= 0) {
+        // If editing an existing task, update its details
+        tasks[editingIndex] = {
+          name: taskDesc,
+          type_tag: taskType,
+          project_tag: taskProject,
+          completed: tasks[editingIndex].completed // Preserve the completed status
+        };
+      } else {
+        // If adding a new task, create a new task object and add to the tasks array
+        var task = {
+          name: taskDesc,
+          type_tag: taskType,
+          project_tag: taskProject,
+          completed: false // New tasks are not completed by default
```

Comment:
Good catch that we need to track whether tasks have been completed haha - could you please update the schema + example in the `data` folder? Thanks!